docs/project_setup_and_build.md

```diff
@@ -0,0 +1,230 @@
+# Project structure and build
+
+## Overview
+
+In this chapter, you will get to know the anatomy of the standard BigFlow project, artifacts, and CLI commands. If you are interested
+in the details of the interface itself, go to [the CLI chapter](./cli.md).
+
+The BigFlow build system packages your processing logic into a Docker image. Thanks to this approach, you can create any
+environment you want for your pipelines. What is more, you don't need to worry about clashes with the Cloud Composer (Airflow) dependencies.
+
+The scheme of a BigFlow artifact looks like this:
+
+![BigFlow artifact scheme](./images/bigflow-artifact.png)
+
+As you can see, your project is turned into a standard Python package (that you can, for example, upload to [pypi](https://pypi.org/) or install locally using `pip`). 
+Next, the package is installed on the Docker image. Finally, there are Airflow DAGs that utilize the produced image.
+
+A single DAG is generated based on the `bigflow.workflow.Workflow` class. 
+Produced DAG consists only of [`KubernetesPodOperator`](https://airflow.apache.org/docs/stable/_api/airflow/contrib/operators/kubernetes_pod_operator/index.html) objects, which
+execute operations on the produced image.
+
+To build a project you need to use the `bigflow build` command. The documentation you are reading is also a valid BigFlow
+project. Go to the `docs` directory and run the `bigflow build` command to see how the build process works. The `bigflow build`
+command should produce three artifacts:
+
+* The `dist` directory with a Python package
```

Comment:
to już jest opisane w cli.md - https://github.com/allegro/bigflow/blob/be0639de64c4e35278c54a6c3d4d94efb46e1391/docs/cli.md#deploying-to-gcp

Ogólnie mamy problem z duplikacją doksów w cli.md vs reszta, trzeba to jakoś przeredagować

Reply:
do przegadania